doc/modules/lmnn.rst

```diff
@@ -80,72 +52,59 @@ in the constructor.
-
+    (x_i - x_l)||^2]_+,
 
-The term inside the sum of the push loss is called the hingle loss and
-the notation amounts to :math:`[x]_+ = \max(0, x)`.
-The parameter :math:`\mu` is a trade-off between penalizing large distances
-to target neighbors and penalizing margin violations by impostors. In practice,
-the two terms are weighted equally.
-
-To use this model for classification, we just need to use a nearest neighbors
-classifier (:class:`classification.KNeighborsClassifier`) on the data
-transformed by the linear transformation :math:`L` found by LMNN.
+where :math:`y_{il} = 1` if :math:`y_i = y_l` and :math:`0` otherwise, :math:`[x]_+ = \max(0, x)` is the hinge loss, and :math:`j \rightsquigarrow i` means that the :math:`j^{th}` sample is a target neighbor of the :math:`i^{th}` sample.
 
-The loss function is not convex in :math:`L`, but there is an alternative
-formulation:
+LMNN solves the following (nonconvex) minimization problem:
 
 .. math::
 
-    \varepsilon_{\text{pull}} (M) = \sum_{i, j \rightsquigarrow i} (x_i-x_j)
-    ^T M (x_i-x_j)
+    \min_L \varepsilon(L) = (1 - \mu) \varepsilon_{\text{pull}} (L) +
+    \mu \varepsilon_{\text{push}} (L) \text{, } \quad \mu \in [0,1].
 
-    \varepsilon_{\text{push}} (M) = \sum_{i, j \rightsquigarrow i}
-    \sum_{l} (1 - y_{il}) [1 + (x_i-x_j)^T M (x_i-x_j) - (x_i-x_l)^T M (x_i-x_l)]_+
-    \text{ where } y_{il} = 1 \text{ if } $y_i = y_l$ \text{ and 0 otherwise}.
+The parameter :math:`\mu` is a trade-off between penalizing large distances
+to target neighbors and penalizing margin violations by impostors. In practice,
+the two terms are weighted equally.
 
-    \varepsilon(M) = (1 - \mu) \varepsilon_{\text{pull}} (M) +
-    \mu \varepsilon_{\text{push}} (M) \text{, } \quad \mu \in [0,1]
+To use this model for classification, one can simply fit a nearest neighbors
+classifier (:class:`KNeighborsClassifier`) on the data
+transformed by the linear transformation :math:`L` found by LMNN (this is implemented by :func:`LargeMarginNearestNeighbor.transform`).
 
 
-This objective is now convex in :math:`M` but the matrix :math:`M = L^T L`
-must be constrained to be positive semidefinite as opposed to :math:`L`
-which could be any real matrix.
+Mahalanobis distance
+--------------------
 
+LMNN can be seen as learning a (squared) Mahalanobis distance metric:
 
-In contrast to related methods such as LDA, LMNN does not make any
-assumptions about the class distributions. The nearest neighbor
-classification can naturally produce highly irregular decision boundaries.
-See [#1]_ for more details.
+.. math::
 
+    || L(x_i - x_j)||^2 = (x_i - x_j)^TM(x_i - x_j),
 
+where :math:`M = L L^T` is a symmetric positive semi-definite matrix of size ``(n_features, n_features)``. The objective function of LMNN can be rewritten and solved with respect to :math:`M` directly. This results in a convex but constrained problem (since :math:`M` must be symmetric positive semi-definite). See [#1]_ for more details.
```

Comment:
Based on the shape of `L` being ``(n_features_out, n_features)``, this should be `M = L^T L`.